src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ThreadPoolTaskScheduler.cs

```diff
@@ -47,7 +47,9 @@ protected internal override void QueueTask(Task task)
                 // Run LongRunning tasks on their own dedicated thread.
                 Thread thread = new Thread(s_longRunningThreadWork);
                 thread.IsBackground = true; // Keep this thread from blocking process shutdown
+#pragma warning disable CA1416 // Validate platform compatibility, Thread.IsThreadStartSupported true means not browser
                 thread.Start(task);
+#pragma warning restore CA1416 // Validate platform compatibility
```

Comment:
I'd have hoped the analyzer would be able to understand the guard clauses like IsThreadStartSupported just as the linker can.  Is that not possible?

Reply:
I am not sure how the linker does that, not sure how the analyzer could understand it, I just saw it will not be set for browser from project settings, but for now, the analyzer not doing anything like that

Reply:
I think we'd need new attributes to decorate properties like `IsThreadStartSupported` to indicate that platform support is implied based on return values.

Would a `Debug.Assert(!OperatingSystem.IsBrowser)` be more appropriate here though, instead of the `#pragma`?

Reply:
> I think we'd need new attributes to decorate properties like IsThreadStartSupported to indicate that platform support is implied based on return values.

I would imagine something like that, was wondering if linker something else instead

> Would a Debug.Assert(!OperatingSystem.IsBrowser) be more appropriate here though, instead of the #pragma?

By suggestions, it is now changed to `#if !TARGET_BROWSER` instead

Reply:
> I'd have hoped the analyzer would be able to understand the guard clauses like IsThreadStartSupported just as the linker can. Is that not possible?

I think I see this scenario the first time, if it is popular enough we can consider support it. I have created an enhancement issue https://github.com/dotnet/roslyn-analyzers/issues/4428